pkgs/development/tools/language-servers/fish-lsp/default.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenv,
+  fetchYarnDeps,
+  fixup-yarn-lock,
+  fetchFromGitHub,
+  yarn,
+  fish,
+  nodejs,
+}: let
+  version = "1.0.6";
+
+  pname = "fish-lsp";
```

Comment:
Is it needed here? This usually indicates a bad pattern.

Reply:
`version` is reuse in the `fetchFromGitHub` and the `mkDerivation` so create a common variable make sense.

`pname` is not reuse anywhere else except in the `mkDerivation` so I can move it directly inside it.